### PR TITLE
remove measurement record block parsing.

### PR DESCRIPTION
**Pull request by longlongyang**

### CODE REVIEW

**Comment by longlongyang** [comment]
fix #309

File: test/spdm-emu/src/secret_impl_sample.rs [hunk 1]
```diff
@@ -29,256 +31,80 @@ fn spdm_measurement_collection_impl(
...
-            digest_value4[..64].copy_from_slice(digest4.data.as_ref());
-            digest_value5[..64].copy_from_slice(digest5.data.as_ref());
-            digest_value6[..64].copy_from_slice(digest6.data.as_ref());
-            digest_value7[..64].copy_from_slice(digest7.data.as_ref());
-            digest_value8[..64].copy_from_slice(digest8.data.as_ref());
-            digest_value9[..64].copy_from_slice(digest9.data.as_ref());
-            digest_value10[..64].copy_from_slice(digest10.data.as_ref());
-
+            let mut record = [0u8; config::MAX_SPDM_MEASUREMENT_VALUE_LEN];
+            record[..0x1E9].copy_from_slice(&[
```
Comment (jyao1) [inline test/spdm-emu/src/secret_impl_sample.rs:48]: please make it clear on how to generate such bytes.

File: spdmlib/tests/test_library.rs [hunk 2]
```diff
@@ -152,40 +173,53 @@ fn test_case0_spdm_measurement_record_structure() {
...
+    assert_eq!(measurement_record.number_of_blocks, 7);
 }
 
 #[test]
-#[should_panic]
 fn test_case1_spdm_measurement_record_structure() {
     let u8_slice = &mut [0u8; 512];
     let mut writer = Writer::init(u8_slice);
+    let mut record = [0u8; config::MAX_SPDM_MEASUREMENT_VALUE_LEN];
+    record[..0x1E9].copy_from_slice(&[
```
Comment (jyao1) [inline spdmlib/tests/test_library.rs:184]: please generate it with code.

File: spdmlib/tests/test_library.rs [hunk 3]
```diff
@@ -125,22 +122,46 @@ fn test_case0_spdm_measurement_record_structure() {
     let u8_slice = &mut [0u8; 512];
     let mut writer = Writer::init(u8_slice);
     SpdmMeasurementRecordStructure::default();
+    let mut record = [0u8; config::MAX_SPDM_MEASUREMENT_VALUE_LEN];
```
Comment (jyao1) [inline spdmlib/tests/test_library.rs:125]: please generate it from

File: spdmlib/src/protocol/algo.rs [hunk 4]
```diff
@@ -825,66 +824,19 @@ pub struct SpdmCertChain {
     pub cert_chain: SpdmCertChainData,
 }
 
-enum_builder! {
-    @U8
-    EnumName: SpdmDmtfMeasurementType;
```
Comment (jyao1) [inline spdmlib/src/protocol/algo.rs:830]: why remove this?

File: spdmlib/src/protocol/algo.rs [hunk 5]
```diff
@@ -825,66 +824,19 @@ pub struct SpdmCertChain {
...
-        SpdmDmtfMeasurementManifest => 0x4,
-        SpdmDmtfMeasurementStructuredRepresentationMode => 0x5,
-        SpdmDmtfMeasurementMutableFirmwareVersionNumber => 0x6,
-        SpdmDmtfMeasurementMutableFirmwareSecurityVersionNumber => 0x7
-    }
-}
-
-enum_builder! {
-    @U8
-    EnumName: SpdmDmtfMeasurementRepresentation;
```
Comment (jyao1) [inline spdmlib/src/protocol/algo.rs:845]: why remove this?

File: sh_script/test_spdm_unit_test_coverage.sh [hunk 6]
```diff
@@ -14,7 +14,7 @@ export LLVM_PROFILE_FILE="your_name-%p-%m.profraw"
 
 cargo build
 
-cargo test 
+cargo test -- --test-threads=2
```
Comment (jyao1) [inline sh_script/test_spdm_unit_test_coverage.sh:17]: what's this?

Comment (longlongyang) [reply sh_script/test_spdm_unit_test_coverage.sh]: > what's this?

already removed.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**